File: java/src/main/java/codingdojo/SpreadsheetException.java
package codingdojo;


import java.util.List;

/**
 * This kind of exception is thrown when something bad
 * happens when evaluating the calculation formulas in a
 * SpreadsheetWorkbook
 */
public class SpreadsheetException extends Exception {
    private List<String> cells;
    private String token;

    public SpreadsheetException(String message, List<String> cells, String token) {
        super(message);
        this.cells = cells;
        this.token = token;
    }

    public List<String> getCells() {
        return cells;
    }

    public String getToken() {
        return token;
    }
}

File: csharp/ExceptionRollup/SpreadsheetException.cs
using System;
using System.Collections.Generic;

namespace codingdojo
{
    public class SpreadsheetException : Exception
    {
        private readonly List<string> _cells;
        private readonly string _token;

        public SpreadsheetException(string message, List<string> cells, string token) : base(message)
        {
            _cells = cells;
            _token = token;
        }

        public List<string> GetCells()
        {
            return _cells;
        }

        public string GetToken()
        {
            return _token;
        }
    }
}